iree/hal/vulkan/vma_allocator.cc

```diff
@@ -188,6 +188,11 @@ static iree_allocator_t iree_hal_vulkan_vma_allocator_host_allocator(
   return allocator->host_allocator;
 }
 
+static iree_status_t iree_hal_vulkan_vma_allocator_trim(
+    iree_hal_allocator_t* base_allocator) {
+  return iree_ok_status();
```

Comment:
Could maybe defrag VMA in here? (not in this PR, but in the future - maybe worth a comment)

https://gpuopen-librariesandsdks.github.io/VulkanMemoryAllocator/html/
https://gpuopen-librariesandsdks.github.io/VulkanMemoryAllocator/html/defragmentation.html
https://gpuopen-librariesandsdks.github.io/VulkanMemoryAllocator/html/staying_within_budget.html

Reply:
That'd be neat but requires a lot more goo - once the defrag completes all buffers that referenced defragged memory need to be recreated. We'll probably switch off VMA before we get around to that - but defragging would be neat to have in our impl!